### PR TITLE
Add CartItem, ShoppingCart entity and missing features

### PR DESCRIPTION
**Pull request by Serhii-Nosulia**

### CODE REVIEW

File: src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                                                          AddBookToShoppingCartDto cartDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(userId).orElseThrow(
+                () -> new EntityNotFoundException("Couldn't find shopping cart by id " + userId)
+        );
+
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            if (cartItem.getBook().getId().equals(cartDto.bookId())) {
+                throw new RuntimeException("This book is already in your cart");
+            }
+        }
```
Comment (olekskov) [inline src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java]: For me it's more logical to increase quantity of book in cart rather then throw exception

File: src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return cartItemMapper.toAddBookToShoppingCartDto(cartItemsRepository.save(cartItem));
+    }
+
+    @Override
+    public SetQuantityDto updateBookQuantity(Long cartItemId, SetQuantityDto setQuantityDto) {
+        CartItem cartItem = cartItemsRepository.findById(cartItemId).orElseThrow(
+                () -> new EntityNotFoundException("Couldn't find cart item by id " + cartItemId)
+        );
+        cartItem.setQuantity(setQuantityDto.quantity());
+        return cartItemMapper.toSetQuantityDto(cartItemsRepository.save(cartItem));
```
Comment (olekskov) [inline src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java]: Better to introduce check that updated cart item is in cart of user that is logged in. Now you may easily change quantity of cart item in other user cart

File: src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public void deleteCartItem(Long id) {
+        cartItemsRepository.deleteById(id);
```
Comment (olekskov) [inline src/main/java/mate/bookshopapp/service/impl/ShoppingCartServiceImpl.java]: same here

File: src/main/java/mate/bookshopapp/repository/shopping/cart/ShoppingCartRepository.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import mate.bookshopapp.model.ShoppingCart;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @Query("FROM shopping_carts sc WHERE sc.id = :id")
+    @EntityGraph(attributePaths = {"cartItems", "cartItems.book"})
+    Optional<ShoppingCart> findShoppingCartById(@Param("id") Long id);
```
Comment (olekskov) [inline src/main/java/mate/bookshopapp/repository/shopping/cart/ShoppingCartRepository.java]: Should work without query
```
    @EntityGraph(attributePaths = {"cartItems", "cartItems.book"})
    Optional<ShoppingCart> findById(Long id);
```

**Review by olekskov — CHANGES_REQUESTED** [review]

**Review by Ivan95kos — APPROVED** [review]
Great job

**Pull request merged**